### PR TITLE
Update video 'database'

### LINKED ISSUE #73: Update "backend" pattern

### PR DESCRIPTION
**Pull request by ChuckBTaylor**
Closes #73

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -6,15 +6,19 @@ import { kurtTheme } from './theme'
...
+  getVideoPageFromVideosPath,
+} from './common/utils/utils'
 
 const router = createBrowserRouter([
   {
     element: <KurtApp />,
     children: [
       {
-        path: '/videos',
+        path: '/videos/:videoPage',
```
Comment (ChuckBTaylor) [inline src/App.tsx:19]: Update route

File: src/assets/pages/directorOfPhotography.ts
Comment (ChuckBTaylor) [inline src/assets/pages/directorOfPhotography.ts:1]: Individual files for each page

File: src/assets/pages/series/lowPheLife.ts
Comment (ChuckBTaylor) [inline src/assets/pages/series/lowPheLife.ts:1]: New Series format

File: src/assets/videos/adHocVideos.ts [hunk 2]
```diff
@@ -28,4 +28,4 @@ export const adHocVideos: IVideoInfo[] = [
     genres: ['documentary'],
     roles: ['director_of_photography', 'director_producer'],
   },
-]
+] as const satisfies Readonly<IVideoInfo[]>
```
Comment (ChuckBTaylor) [inline src/assets/videos/adHocVideos.ts:31]: `satisfies` gives us to ability to type check while maintaining the `const` values inside each object

File: src/assets/videos/index.ts [hunk 3]
```diff
@@ -1,6 +1,12 @@
...
-import { adHocVideos } from './videos'
+import { adHocVideos } from './adHocVideos'
 
-const videos = [adHocVideos, lowPheLifeVideos].flatMap(i => i)
+const videos = [
+  ...adHocVideos,
+  ...lowPheLifeVideos,
+] as const satisfies Readonly<IVideoInfo[]>
+
+export type VideoTitle = (typeof videos)[number]['title']
```
Comment (ChuckBTaylor) [inline src/assets/videos/index.ts:10]: Type that is all the video titles in `videos`

File: src/common/constants/constants.ts [hunk 4]
```diff
@@ -1,10 +1,15 @@
...
+import { documentaryPageInfo } from '../../assets/pages/documentary'
+import { IVideoListPageInfo } from '../interfaces/IVideoListPageInfo'
+import { Genre, Role } from './enums'
 
 export type PageEnum = Genre | Role
 
 interface EnumMapping {
   label: string
   url: string
+  pageInfo?: IVideoListPageInfo
```
Comment (ChuckBTaylor) [inline src/common/constants/constants.ts:12]: Point enum at pageInfo

File: src/common/interfaces/ISeriesDetail.ts [hunk 5]
```diff
@@ -1,11 +1,12 @@
...
 import { Genre, Role } from '../constants/enums'
 
 export interface ISeriesDetail {
   path: string // /series/{path}
   name: string
   imageUrl?: string
-  orderedVideoUrls: string[]
-  highlightUrl?: string // used on series list page
+  orderedVideoTitles: VideoTitle[]
+  highlightTitle?: VideoTitle // used on series list page
```
Comment (ChuckBTaylor) [inline src/common/interfaces/ISeriesDetail.ts:9]: Use titles instead of URLs

File: src/components/atoms/pageTitle/PageTitle.tsx [hunk 6]
```diff
@@ -2,14 +2,31 @@ import { Container, SxProps, Typography } from '@mui/material'
 
 type PageTitleProps = {
   title: string
+  titleUnderlineColor?: string
```
Comment (ChuckBTaylor) [inline src/components/atoms/pageTitle/PageTitle.tsx:5]: Some titles have a colored bar underneath. Add optional prop fore that color. If color is present, render bar

File: src/components/templates/VideoListPage.tsx [hunk 7]
```diff
@@ -1,70 +1,37 @@
...
-  }
-
-  const sanitzedValue = queryValue.replaceAll('-', '_')
-
-  if (!roleTypeChecker(sanitzedValue) && sanitzedValue !== 'documentary') {
-    return null
-  }
-
-  return sanitzedValue
-}
```
Comment (ChuckBTaylor) [inline src/components/templates/VideoListPage.tsx:59]: No longer need to calculate from route. Loader function is passing all values in

**Review by mwilkins74 — APPROVED** [review]

**Pull request merged**